src/components/SourcesTree.js

```diff
@@ -27,32 +28,47 @@ type CreateTree = {
   highlightItems?: any
 };
 
-let SourcesTree = React.createClass({
-  propTypes: {
-    sources: ImPropTypes.map.isRequired,
-    selectSource: PropTypes.func.isRequired,
-    shownSource: PropTypes.string,
-    selectedSource: ImPropTypes.map
-  },
-
-  displayName: "SourcesTree",
+class SourcesTree extends Component {
+  state: CreateTree
+  focusItem: Function
+  selectItem: Function
+  getIcon: Function
+  queueUpdate: Function
+  onContextMenu: Function
+  renderItem: Function
+  mounted: bool
+
+  constructor(props) {
+    super(props);
+    this.state = createTree(this.props.sources);
+
+    this.focusItem = this.focusItem.bind(this);
+    this.selectItem = this.selectItem.bind(this);
+    this.getIcon = this.getIcon.bind(this);
```

Comment:
ES6 classes doesn't correctly bind context to function, so `.bind(this)` is used.
https://coderwall.com/p/hdklna/the-right-way-to-bind-custom-methods-in-es6-class-react-component